src/test/java/org/myrobotlab/service/HttpClientTest.java

```diff
@@ -0,0 +1,99 @@
+package org.myrobotlab.service;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.myrobotlab.framework.Message;
+import org.myrobotlab.framework.Service;
+import org.myrobotlab.service.data.HttpData;
+
+public class HttpClientTest extends
+    AbstractServiceTest /* implements HttpDataListener - not useful .. "yet" */ {
+
+  TestCatcher catcher;
+
+  @Before
+  public void setUp() throws Exception {
+    catcher = (TestCatcher) Runtime.start("catcher", "TestCatcher");
+  }
+
+  @Override
+  public Service createService() {
+    return (Service) Runtime.start("http", "HttpClient");
+  }
+
+  /**
+   * publishHttpData contains more information content type, response code,
+   * etc... need to subscribe to it manually for testing purposes
+   * 
+   */
+  /*
+   * @Override //FIXME - add blocking queue - to verify public void
+   * onHttpData(HttpData data) { log.info(data.toString()); queue.add(data); }
+   */
+
+  /**
+   * for testing purposes
+   * 
+   */
+  /*
+   * @Override //FIXME - add blocking queue - to verify public void
+   * onHttpResponse(String data) { log.info(data); queue.add(data); }
+   */
+
+  @Override
+  public void testService() throws Exception {
+    HttpClient http = (HttpClient) Runtime.start("http", "HttpClient");
```

Comment:
this should be created in the createService() method.  this.service is the httpClient that you created already.

Reply:
I'm not sure I understand the benefit of creating services in  @Before setup although we do it all the time.   

I think potentially there is a benefit of starting all services necessary at the beginning of a test.  The test is potentially fully atomic ... no special setup needs to be run in order to do the test, yet if this used the static reference - the method would throw an NPE

I removed the reference in setup for the reasons above.

AbstractTest already did releasing of services, additionally it keeps track of warrant threads and reports on them.

So, I ended up removing everything except the test itself.